src/core/core/rpc_engine.cpp

```diff
@@ -154,29 +154,16 @@ namespace dsn {
             // server address side effect
             auto req = call->get_request();
             auto sp = task_spec::get(req->local_rpc_code);
-            if (reply->from_address != req->to_address)
+            // TODO(qinzuoyan): because current rpc forward is fake (resend from client), we cannot
```

Comment:
ok, I'll fix it. Actually, the reply->error() must be ERR_FORWARD_TO_OTHERS or ERR_OK here.